File: dcp-solutions/2024-12-17/solution.rb
```ruby
def exist(board, word)
  m, n = board.size, board[0].size
  visited = Array.new(m) { Array.new(n, false) }
  
  def dfs(i, j, k)
    return true if k == word.length
    return false if i < 0 || j < 0 || i >= m || j >= n || !board[i][j] || visited[i][j]
    
    temp = board[i][j]
    board[i][j] = '#'
    visited[i][j] = true
    
    if dfs(i + 1, j, k + 1) || dfs(i - 1, j, k + 1) || dfs(i, j + 1, k + 1) || dfs(i, j - 1, k + 1)
      return true
    end
    
    board[i][j] = temp
    visited[i][j] = false
  end
  
  (0...m).each do |i|
    (0...n).each do |j|
      dfs(i, j, 0) if word[0] == board[i][j]
    end
  end
  
  !word.empty?
end
```

File: dcp-solutions/2024-12-17/solution.php
```php
function exist($board, $word) {
    if (empty($board)) return false;
    $m = count($board);
    $n = count($board[0]);
    $directions = [[-1, 0], [1, 0], [0, -1], [0, 1]];
    
    function dfs(&$board, $i, $j, $word, &$directions) {
        if (empty($word)) return true;
        
        if ($i < 0 || $i >= count($board) || $j < 0 || $j >= count($board[0]) || $word[0] != $board[$i][$j]) {
            return false;
        }
        
        $tmp = $board[$i][$j];
        $board[$i][$j] = '#';
        
        foreach ($directions as $direction) {
            if (dfs($board, $i + $direction[0], $j + $direction[1], substr($word, 1), $directions)) return true;
        }
        
        $board[$i][$j] = $tmp;
        return false;
    }
    
    for ($i = 0; $i < count($board); ++$i) {
        for ($j = 0; $j < count($board[0]); ++$j) {
            if (dfs($board, $i, $j, $word, $directions)) return true;
        }
    }
    
    return false;
}
```